pkg/kubelet/kubelet_node_status.go

```diff
@@ -554,3 +568,56 @@ func validateNodeIP(nodeIP net.IP) error {
 	}
 	return fmt.Errorf("Node IP: %q not found in the host's network interfaces", nodeIP.String())
 }
+
+// nodeStatusHasChanged compares the original node and current node's status and
+// returns true if any change happens. The heartbeat timestamp is ignored.
+func nodeStatusHasChanged(originalStatus *v1.NodeStatus, status *v1.NodeStatus) bool {
+	if originalStatus == nil || status == nil {
+		return !apiequality.Semantic.DeepEqual(originalStatus, status)
+	}
+
+	originalStatusElem := reflect.ValueOf(originalStatus).Elem()
+	statusElem := reflect.ValueOf(status).Elem()
```

Comment:
In general I'm not a fan of reflection if it's possible to avoid that.
How about something like that:

```
if nodeConditionsHaveChanged(originalStatus.Conditions, status.Conditions) {
  return true
}
originalConditions, conditions := originalStatus.Conditions. status.Conditions
originalStatus.Conditions = nil
status.Conditions = nil
defer func() {
  originalStatus.Conditions = originalConditions
  status.Conditions = conditions
}
return !apiequality.Semantic.DeepEqual(originalStatus, status)
```

Reply:
Done.

Just curious, why is reflection not preferred?

Reply:
reflection is really slow - this is the reason why we autogenerate a bunch of code